topics/icrc_32_sign_challenge.md

```diff
@@ -0,0 +1,126 @@
+
+## Request
+
+**`version` (`text`):** The version of the standard used. If the signer does not support the version of the request, it must send the `"VERSION_NOT_SUPPORTED"` error in response.
+
+**`challenge` (`blob`):** A challenge used for the signer to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated from a secure random source by the sender of the request.
+
+### Example RPC Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc32_sign_challenge",
+  "params": {
+    "version": "1",
+    "principal": "rwlgt-iiaaa-aaaaa-aaaaa-cai",
+    "challenge": "UjwgsORvEzp98TmB1cAIseNOoD9+GLyN/1DzJ5+jxZM="
+  }
+}
+```
+
+## Response
+
+`version` (`text`): The version of the standard used. It must match the `version` from the request.
+
+`identities`: A list of identities the user has selected to share with the relying party.
+
+- `publicKey` (`blob`): The DER-encoded public key associated with the identity, derived in accordance with one of [the signature algorithms supported by the IC](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures). The public key can be used to [derive a self-authenticating principal](https://internetcomputer.org/docs/current/references/ic-interface-spec/#principal).
+- `signature` (`blob`): The signature produced by signing the concatenation of the domain separator `\x13ic-signer-challenge` (UTF-8 encoded) and the challenge with the private key associated with the identity.
+- `delegation` (optional): An array of delegations (as defined by the [IC interface specification, authentication section](https://internetcomputer.org/docs/current/references/ic-interface-spec/#authentication)):
+  - object with the following properties:
+    - `pubkey` (`blob`): Public key as described in the [IC interface specification, signatures section](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures).
+    - `expiration` (`text`): Expiration of the delegation, in nanoseconds since 1970-01-01, as a base-10 string.
+    - `targets` (`text` array): A list of target canister ids (textual representation) the delegation is restricted to making canister calls to. If the list is not present, the delegation applies to all canisters (i.e. it is not restricted).
+  - `signature` (`blob`): Signature on the 32-byte representation-independent hash of the map contained in the delegation field as described in [IC interface specification, signatures section](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures), using the 27 bytes `\x1Aic-request-auth-delegation` as the domain separator. 
+
+### Example RPC Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "version": "1",
+    "identities": [
+      {
+        "publicKey": "MIIBIjANBgkqhkiG",
+        "signature": "MEUCIQDQ8Z3Z"
+      }
+    ]
+  }
+}
+```
+
+## Message Processing
```

Comment:
This section needs updating:
- wrong method name
- there is no identity selection anymore (as the RP chooses the identity) and only a single key is used to create s single signature
- step 5 needs to be updated to reflect that the response does no longer contain a list and that the RP needs to check that the signature corresponds to the identity specified in the request